### PR TITLE
Added support for UPC routers, r42, v3.15.0.1

### PR DESCRIPTION
**Pull request by ph4r05**
UPC routers support added
Based on https://haxx.in/upc_keys.c

### CODE REVIEW

**Comment by ph4r05** [comment]
Check it out now

**Comment by ph4r05** [comment]
If you have some suggestion feel free to contact me

**Comment by ph4r05** [comment]
I have modified UPC native keygen so it supports progress monitoring and passing computed keys to java right away (no need to allocate memory for computed keys before computation).

File: android/routerKeygen/src/main/jni/upc/src/upc_keys.c [hunk 1]
```diff
@@ -0,0 +1,169 @@
...
+                    "  by blasty <peter@haxx.in>\n\n"
+    );
+}
+
+void usage(char *prog)
+{
+    printf("Usage: %s <ESSID>\n", prog);
+}
+
+int original_main(int argc, char *argv[])
```
Comment (ruiaraujo) [inline android/routerKeygen/src/main/jni/upc/src/upc_keys.c]: Could you remove this function?

File: android/routerKeygen/src/main/AndroidManifest.xml [hunk 2]
```diff
@@ -2,8 +2,8 @@
 <manifest xmlns:android="http://schemas.android.com/apk/res/android"
     package="org.exobel.routerkeygen"
     android:installLocation="auto"
-    android:versionCode="39"
-    android:versionName="3.15.0">
+    android:versionCode="42"
+    android:versionName="3.15.0.1">
```
Comment (ruiaraujo) [inline android/routerKeygen/src/main/AndroidManifest.xml]: Please dont touch this file on this PR.

**Comment by ruiaraujo** [comment]
Please just address the last few comments and I will merge this.
Also could you please rebase this PR into a single commit ?

**Comment by ph4r05** [comment]
I've addressed the comments, check it out now pls.

**Comment by ruiaraujo** [comment]
Thanks for the great job.

**Pull request merged**